### ISSUE TITLE
Model FirstOrNew(), a bit update

### ISSUE THREAD

**Issue by raiika** [laravel/ideas#1246]
public function firstOrNew(array $values = [], array $attributes = [])
    {
        if (! is_null($instance = $this->where($attributes)->first())) {
            return $instance;
        }
 
        return $this->newModelInstance($attributes + $values);
    }

Thus, we could call like this

$model->firstOrNew();
$model->firstOrNew($request->input());
$model->firstOrNew($request->input(), ['child_id' => 1]);

**Comment by sisve** [comment]
Isn't this same as `$model->first() ?? new Model($request->input())`?

**Comment by raiika** [comment]
yes it is, if that is the one that I should use, what is the use of this firstOrNew method ?
